libraries/Microsoft.Bot.Builder.Dialogs/Choices/ChoiceRecognizers.cs

```diff
@@ -80,7 +81,11 @@ private static void MatchChoiceByIndex(IList<Choice> list, List<ModelResult<Foun
         {
             try
             {
-                var index = int.Parse(match.Resolution.Value) - 1;
+                var value = match.Resolution.Value.Replace("end", list.Count.ToString());
+                DataTable dt = new DataTable();

```

Comment:
Please explain what this is doing with a comment.

Reply:
Done. Thanks!